src/lib/parsers/include/sol-json.h

```diff
@@ -697,20 +696,20 @@ char *sol_json_escape_string(const char *str, char *buf, size_t len) SOL_ATTR_NO
  * @param s String to escape
  */
 #define SOL_JSON_ESCAPE_STRING(s) ({ \
-        size_t str_len ## __COUNT__  = sol_json_calculate_escaped_string_len(s); \
-        sol_json_escape_string(s, (char *)alloca(str_len ## __COUNT__), str_len ## __COUNT__); \
+        SOL_BUFFER_DECLARE_STATIC(aux_buf ## __COUNT__, strlen(s) + 1); \
```

Comment:
why are you using `strlen(s) + 1`? The buffer should be of the size of the escaped length, not the input length. Like escaping `a\n` (2 bytes)  will be `a\\n` (3 bytes).